packages/messaging/src/index.ts

```diff
@@ -195,6 +195,29 @@ export type MessageHook =
  * The namespace for the global singleton message loop.
  */
 export namespace MessageLoop {
+  /**
+   * A function that cancels the pending loop task; `null` if unavailable.
+   */
+  let pending: (() => void) | null = null;
+
+  /**
+   * Schedules a function for invocation as soon as possible asynchronously.
+   *
+   * @param fn The function to invoke when called back.
+   *
+   * @returns An anonymous function that will unschedule invocation if possible.
+   */
+  const schedule = (
+    resolved =>
+    (fn: () => unknown): (() => void) => {
+      let rejected = false;
+      resolved.then(() => !rejected && fn());
+      return () => {
+        rejected = true;
+      };
+    }
+  )(Promise.resolve());
```

Comment:
As mentioned in last weeks lab dev meeting, using a Promise or async will schedule the task as a microtask, compared to `requestAnimationFrame` / `setTimeout` etc that will schedule it as a macro task. This difference is likely to be significant (@afshin has the details).

Reply:
It is safe in this circumstance because message propagation is both lightweight and synchronous. This is the reason why the same technique is unavailable to us in `Poll`.

Reply:
> @afshin should we move forward with this one?

Yes, I wanted to give @vidartf a chance to look, but I'm happy to move ahead.